### PR TITLE
Feat/add local debugging support

### PR DESCRIPTION
**Pull request by cnlangzi**
Tasks to complete before merging PR:
- [x]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [x]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [x]  Merge your system tests PR to master BEFORE merging this PR  
  
Associated PRs (Link as appropriate):
...

changes:
- added dev.local/cli.sh to support debugging outside docker
  + install deps
  + install cassandra
  + install postgres
  + install redis/redis_txns
  + build and run sharder/miner
  + install debugger config in vscode

### CODE REVIEW

File: code/go/0chain.net/miner/miner/main.go [hunk 1]
```diff
@@ -42,16 +43,30 @@ func main() {
 	delayFile := flag.String("delay_file", "", "delay_file")
 	magicBlockFile := flag.String("magic_block_file", "", "magic_block_file")
 	initialStatesFile := flag.String("initial_states", "", "initial_states")
+
+	workdir := ""
```
Comment (ssardana96) [inline code/go/0chain.net/miner/miner/main.go]: for neat we can create all var in beginning
var (
workdir string
redisHost string
redisPort int
.
.
.

)

File: dev.local/cli.base.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+#clean all data
+cleanAll() {
+    cd $root
```
Comment (ssardana96) [inline dev.local/cli.base.sh:5]: whats the value of $root ?

**Review by ssardana96 — CHANGES_REQUESTED** [review]

**Comment by ssardana96** [comment]
lgtm
just few neats.

File: dev.local/cli.base.sh (hunk 2 above)
Comment (cnlangzi) [reply dev.local/cli.base.sh:5]: it is ./dev.local

Comment (kushthedude) [reply dev.local/cli.base.sh:5]: It is defined in the files referencing this function, it is just the present working directory, https://github.com/0chain/0chain/blob/11d3c0f916ea53c84b0faea15ab1c74ef50bd829/dev.local/cli.sh#L5

**Review by kushthedude — COMMENTED** [review]
Can we add a CI Step to try running this complete script? If we are going to shift to automated scripts to run check, I recommend we should have a sanity check that if the script is always working or not, which in my opinion was one of the major pitfalls of manually setting up the chain since no-one actually had the bandwidth to try these steps manually everytime with each commit.

**Comment by cnlangzi** [comment]
> Can we add a CI Step to try running this complete script? If we are going to shift to automated scripts to run check, I recommend we should have a sanity check that if the script is always working or not, which in my opinion was one of the major pitfalls of manually setting up the chain since no-one actually had the bandwidth to try these steps manually everytime with each commit.

it is designed for local development. developer runs it everyday like I do in blobber development. it seems be not worth to check it in CI job with each commit. and it is an interactive script only for MacOS.

**Review by ssardana96 — APPROVED** [review]

File: code/go/0chain.net/miner/miner/main.go [hunk 3]
```diff
@@ -36,6 +36,14 @@ import (
 	"0chain.net/smartcontract/setupsc"
 )
 
+var (
+	workdir       string
```
Comment (ssardana96) [inline code/go/0chain.net/miner/miner/main.go]: @cnlangzi you can keep them local

Comment (ssardana96) [reply code/go/0chain.net/miner/miner/main.go]: Local to main

**Review by ssardana96 — CHANGES_REQUESTED** [review]

**Review by ssardana96 — APPROVED** [review]
good to merge

**Pull request merged**